numpyro/distributions/constraints.py

```diff
@@ -428,6 +428,7 @@ def __call__(self, x):
         pad_width = [(0, 0)] * x.ndim
         pad_width[-1] = (0, 1)
         z_padded = np.pad(z, pad_width, mode="constant", constant_values=1.)
+        pad_width = [(0, 0)] * x.ndim
```

Comment:
This is needed because reusing the same `pad_width` results in a cache hit (the id for the arg doesn't change) and the older value of the function call is used.